interfaces/Input_to_core.atd

```diff
@@ -18,13 +18,15 @@
  * tests/e2e/target
 *)
 
-type target = {
+type xlang <ocaml attr="deriving show"> = string wrap <ocaml module="Xlang">
+
+type target <ocaml attr="deriving show"> = {
   path: string (* source file *);
-  (* examples: "ocaml", "python", but also "generic" or "regexp".
-   * less: would be great to use Lang.t here, but atd does not
-   * allow (by design) to use types defined in other files.
-   *)
-  language: string;
+  (* Must be a valid "language" as defined in Xlang.mli.
+     examples: "ocaml", "python", but also "spacegrep" or "regexp".
+     less: would be great to use Lang.t here, but atd does not
+     allow (by design) to use types defined in other files. *)
```

Comment:
I stopped because it got unreasonably complicated to add to atdgen, while it was rather straightforward to add to atdpy. This is when I decided we should rewrite atdgen from scratch using the same model as atdpy or atdts:
* this is the modules PR, which works fine for atdpy and atdts but contains massive unfinished refactoring for atdgen that was an attempt to understand the code better: https://github.com/ahrefs/atd/pull/297. I might merge the atdpy/atdts implementation first, after which we'll have imports for Python and TypeScript but not for OCaml...
* issue pushing for an atdml: https://github.com/ahrefs/atd/issues/313